drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -405,15 +417,23 @@ local function find_default_endpoint(device)
   return device.MATTER_DEFAULT_ENDPOINT
 end
 
-local function assign_child_profile(device, child_ep)
+local function assign_child_profile(device, child_ep, ep_sequence)
   local profile
 
   -- check if device has an overridden child profile that differs from the profile
   -- that would match the child's device type
   for _, fingerprint in ipairs(child_device_profile_overrides) do
     if device.manufacturer_info.vendor_id == fingerprint.vendor_id and
        device.manufacturer_info.product_id == fingerprint.product_id then
-      return fingerprint.child_profile
+      if device.manufacturer_info.vendor_id == AQARA_MANUFACTURER_ID then
+        if ep_sequence == 1 then
+          -- To add Electrical Sensor only to the first EDGE_CHILD
+          device:set_field(IS_AQARA_SWITCH_DEVICE, true)
+          return fingerprint.child_profile
+        end
+      else
+        return fingerprint.child_profile
+      end
```

Comment:
```suggestion
        end
        return fingerprint.child_profile
```
There's no reason to overcomplicate this with multiple returns. There just needs to be extra logic before the return for the aqara devices.

Reply:
If two Light On/Off devices are added as EDGE_CHILD, only the first device is defined as `fingerprint.child_profile` (`light-power-energy-powerConsumption` profile), and the profile of the second device is to select the `light-binary` profile by the logic below.
Therefore, we need to keep the existing logic as it is, so I added comments.

Reply:
sorry, what I meant here is that we don't need to do and early return. both paths end in the call `fingerprint.child_profile`, so we don't need and else statement like that. Both paths return the same thing at the end.

Reply:
@hcarter-775 I'm sorry. If there is anything I still misunderstand, please tell me again.
This is because if `ep_sequence` is `2` or more, `fingerprint.child_profile` should not be returned.
In other words, in the second `EDGE_CHILD` device with `ep_sequence` of `2`, the `light-binary` profile must be specified, so it is implemented as above. Is this part wrong?

Reply:
please see [this message](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1822/files#r1917200411)